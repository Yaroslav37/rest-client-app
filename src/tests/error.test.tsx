import { fireEvent, render, screen } from '@testing-library/react';
import { beforeEach, describe, expect, it, vi } from 'vitest';

import Error from '@/app/error';

vi.mock('next-intl', () => ({
  useTranslations: vi.fn((namespace: string) => {
    return (key: string) => {
      const translations: Record<string, Record<string, string>> = {
        Error: {
          error: 'Something went wrong!',
          try: 'Try again',
        },
      };
      return translations[namespace]?.[key] || key;
    };
  }),
}));

describe('Error', () => {
  const mockError = {
    message: 'Test error message',
    name: 'TestError',
    digest: '12345',
  };

  const mockReset = vi.fn();

  beforeEach(() => {
    vi.clearAllMocks();
    console.error = vi.fn();
  });

  it('renders the error message and try again button', () => {
    render(<Error error={mockError} reset={mockReset} />);

    expect(screen.getByText('Something went wrong!')).toBeInTheDocument();
    expect(screen.getByText('Try again')).toBeInTheDocument();
  });

  it('calls console.error with the error', () => {
    render(<Error error={mockError} reset={mockReset} />);

    expect(console.error).toHaveBeenCalledWith(mockError);
  });

  it('calls reset function when button is clicked', () => {
    render(<Error error={mockError} reset={mockReset} />);

    const button = screen.getByText('Try again');
    fireEvent.click(button);

    expect(mockReset).toHaveBeenCalledTimes(1);
  });

  it('applies correct styling classes', () => {
    const { container } = render(<Error error={mockError} reset={mockReset} />);

    expect(container.firstChild).toHaveClass('min-h-screen');
    expect(container.firstChild).toHaveClass('bg-[var(--color-dark)]');
    expect(screen.getByText('Something went wrong!')).toHaveClass('text-light-green');
    expect(screen.getByText('Try again')).toHaveClass('bg-grey');
  });

  it('matches snapshot', () => {
    const { asFragment } = render(<Error error={mockError} reset={mockReset} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
