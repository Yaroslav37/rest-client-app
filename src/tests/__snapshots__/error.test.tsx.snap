// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Error > matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="min-h-screen flex flex-col items-center justify-center bg-[var(--color-dark)] text-white p-4"
  >
    <h2
      class="text-light-green text-[16px] xs:text-[24px] mb-4"
    >
      Something went wrong!
    </h2>
    <button
      class="
    bg-grey
    text-white
    px-4 py-2
    rounded-lg
    transition-colors
    duration-300
    hover:bg-light-green
    cursor-pointer
      hover:scale-105
      transform
      active:scale-95
  "
    >
      Try again
    </button>
  </div>
</DocumentFragment>
`;
